src/sapling.rs

```diff
@@ -0,0 +1,27 @@
+//! Signature types for the Sapling protocol.
+
+use super::SigType;
+
+/// A type variable corresponding to Zcash's Sapling `SpendAuthSig`.
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum SpendAuth {}
+// This should not exist, but is necessary to use zeroize::DefaultIsZeroes.
+impl Default for SpendAuth {
+    fn default() -> Self {
+        unimplemented!()
+    }
+}
```

Comment:
This is necessary because a `DefaultIsZeroes: Default` bound on `Foo<S>` requires `S: Default`. Per the comment in the commit message for https://github.com/ZcashFoundation/redjubjub/pull/63/commits/2745d378ab35e7bc3268e03d463c7b3c3f8e1cc7 we could replace `DefaultIsZeroes` with explicit `Zeroize` and `Drop` derives; I think we should do that (because having a panicking `Foo::default()` on an unconstructable type will create a Bad Time for someone.

Reply:
I created an issue to do this later https://github.com/ZcashFoundation/reddsa/issues/1